### PR TITLE
support for Membrane users

### PR DESCRIPTION
**Pull request by skurfer**
These are some small fixes we made (and one not so small) to get things working with Membrane users.

### CODE REVIEW

File: collective/pwexpiry/patches.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 from plone.app.users.browser.register import BaseRegistrationForm
 from Products.CMFCore.utils import getToolByName
 from Products.CMFPlone.RegistrationTool import RegistrationTool
+from Products.membrane.plugins.usermanager import MembraneUserManager
```
Comment (frisi) [inline collective/pwexpiry/patches.py]: this will break instances w/o Products.membrane installed

File: collective/pwexpiry/patches.py [hunk 2]
```diff
@@ -150,3 +151,45 @@ def authenticateCredentials(self, credentials):
 ZODBUserManager.authenticateCredentials = authenticateCredentials
 logger.info("Patching Products.PluggableAuthService.plugins.ZODBUserManager."
             "ZODBUserManager.authenticateCredentials")
+
+
+MembraneUserManager.original_authenticateCredentials = MembraneUserManager.authenticateCredentials
```
Comment (frisi) [inline collective/pwexpiry/patches.py]: please use collective.monkeypatcher for adding new patches

File: collective/pwexpiry/pwexpiry_plugin.py [hunk 3]
```diff
@@ -76,7 +76,7 @@ def authenticateCredentials(self, credentials):
             return None
 
         password_date = user.getProperty('password_date', '2000/01/01')
-        if str(password_date) != '2000/01/01':
+        if password_date != DateTime('2000/01/01'):
```
Comment (frisi) [inline collective/pwexpiry/pwexpiry_plugin.py:79]: as we are storing passwort_date with type date in `memberdata_properties.xml`

``` xml
<property name="password_date" type="date">2000/01/01</property>
```

this looks more sane to me

unfortunately we don't have any tests yet

File: collective/pwexpiry/scripts/notify_and_expire.py [hunk 4]
```diff
@@ -55,7 +55,7 @@ def notify_and_expire():
         notifications_to_use = registry['collective.pwexpiry.notification_actions']
     current_time = portal.ZopeTime()
     local_tz = current_time.timezone()
-    for user_id in portal.acl_users.source_users.getUserIds():
+    for user_id in portal.acl_users.getUserIds():
```
Comment (frisi) [inline collective/pwexpiry/scripts/notify_and_expire.py:58]: don't know enough about acl_users.
can you explain why this change is necessary and if it will work on installations w/o membrane too.

File: collective/pwexpiry/patches.py (hunk 1 above)
Comment (skurfer) [inline collective/pwexpiry/patches.py]: Good point. We’ll look at a better way.
